crates/modelardb_server/src/query/grid_exec.rs

```diff
@@ -373,3 +390,64 @@ impl RecordBatchStream for GridStream {
         self.schema.clone()
     }
 }
+
+/// Metrics collected by [`GridStream`] for use by EXPLAIN ANALYZE.
+#[derive(Debug)]
+struct GridStreamMetrics {
+    /// Default set of metrics collected by operators.
+    baseline_metrics: BaselineMetrics,
+    /// Number of data points reconstructed from segments.
+    created_rows: Count,
+    /// Number of data points reconstructed from segments grouped by model type.
+    created_rows_by_model_type: [Count; 3],
+    /// Number of segments with residuals.
+    segments_with_residuals: Count,
+    /// Number of segments grouped by model type.
+    segments_with_model_type: [Count; 3],
```

Comment:
Since segments equals rows this is already tracked for all operators by [`BaselineMetrics`](https://docs.rs/datafusion/34.0.0/datafusion/physical_plan/metrics/struct.BaselineMetrics.html) as shown [here](https://docs.rs/datafusion-physical-plan/34.0.0/src/datafusion_physical_plan/metrics/baseline.rs.html#47).